mysql-test/suite/galera/t/MDEV-25039.test

```diff
@@ -0,0 +1,73 @@
+#
+#  MDEV-25039: MDL BF-BF conflict because of foreign key
+#
+
+--source include/galera_cluster.inc
+--source include/have_innodb.inc
+--source include/have_debug_sync.inc
+--source include/galera_have_debug_sync.inc
+
+# node 2 control connection
+--connect node_1_ctrl, 127.0.0.1, root, , test, $NODE_MYPORT_1
+--connection node_1_ctrl
+SET SESSION wsrep_sync_wait=0;
+
+# node 2 control connection
+--connect node_2_ctrl, 127.0.0.1, root, , test, $NODE_MYPORT_2
+--connection node_2_ctrl
+SET SESSION wsrep_sync_wait=0;
+
+--connection node_1
+
+CREATE TABLE parent(f1 INT NOT NULL PRIMARY KEY, f2 INT);
+CREATE TABLE child(f1 INT NOT NULL PRIMARY KEY, f2 INT, fk INT, FOREIGN KEY(fk) REFERENCES parent(f1));
+
+INSERT INTO parent VALUES (1, 1), (2, 2);
+INSERT INTO child VALUES (1, 1, 1), (2, 2, 2);
+
+## BF - BF conflict // node2
+
+--connection node_1_ctrl
+# Setup galera debug sync before replications for DML (INSERT)
+--let $galera_sync_point = before_replicate_sync
+--source include/galera_set_sync_point.inc
+
+--connection node_1
+--send INSERT INTO parent VALUES (3, 3);
+
+--connection node_1_ctrl
+# Setup galera debug sync before replications for DDL (CREATE INDEX)
+--let $galera_sync_point = before_replicate_sync
+--source include/galera_wait_sync_point.inc
+
+--connection node_2_ctrl
+# First let's wait that nodes reaches debug_sync and 
+# than execute INSERT INTO parent table
+--let $galera_sync_point = before_replicate_sync
+--source include/galera_set_sync_point.inc
+
+--connection node_2
+--send CREATE INDEX idx ON child (f2, fk);
+
+--connection node_2_ctrl
+# Wait for DDL (CREATE INDEX) reach debug sync point
+# Than signal and clear debug points
+--let $galera_sync_point = before_replicate_sync
+--source include/galera_wait_sync_point.inc
+--source include/galera_signal_sync_point.inc
+--source include/galera_clear_sync_point.inc
+
```

Comment:
the sync point before_replicate_sync probably stops the execution before the replication happens, but which of the sessions node_1 or node_2 is supposed to replicate first? or does it not matter? in this test, there is no guarantee of the replication order